go-controller/pkg/ovn/gateway_init.go

```diff
@@ -21,6 +25,67 @@ import (
 	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
 )
 
+// cleanupStalePodSNATs removes SNATs against nodeIP for the given node if the SNAT.logicalIP isn't an active podIP on this node
+// We don't have to worry about missing SNATs that should be added because addLogicalPort takes care of this for all pods
+// when RequestRetryObjs is called for each node add.
+// This is executed only when disableSNATMultipleGateways = true
+//
+// NOTE: On startup libovsdb adds back all the pods and this should normally update all existing SNATs
+// accordingly. Due to a stale egressIP cache bug https://issues.redhat.com/browse/OCPBUGS-1520 we ended up adding
+// wrong pod->nodeSNATs which won't get cleared up unless explicitly deleted.
+// NOTE2: egressIP SNATs are synced in EIP controller.
+func (oc *Controller) cleanupStalePodSNATs(nodeName string, nodeIPs []*net.IPNet) error {
+	if !config.Gateway.DisableSNATMultipleGWs {
+		return nil
```

Comment:
if for some reason someone switched modes here and the flag was false, we would have an SNAT for the entire subnet, and there could be leftover per pod SNATs. In that case we should clean that up. Can be a TODO item here with a github issue to fix later:

search for any snat entries, with a /32 or /128 and external ip == node ip, then delete

Reply:
ack good point, opened this https://github.com/ovn-org/ovn-kubernetes/issues/3232 . will add a TODO in code and fix it in a FUP.